code/sys/sys_unix.c

```diff
@@ -552,7 +553,10 @@ void Sys_Sleep( int msec )
 		if( msec < 0 )
 			msec = 10;
 
-		usleep( msec * 1000 );
+		struct timespec req;
+		req.tv_sec = msec/1000;
+		req.tv_nsec = (msec%1000)*1000000;
```

Comment:
I don't think that you want to be setting BOTH `tv_sec` AND `tv_nsec`.  I think you justwant to be setting `tv_sec`?

Reply:
Based on https://stackoverflow.com/a/7684384, I think you do have to set both given that `tv_nsec` is limited to the range [0,999999999]